skimage/measure/_polygon.py

```diff
@@ -68,9 +93,9 @@ def approximate_polygon(coords, tolerance):
                               projected_lengths1 > 0)
         eucl = np.logical_not(perp)
         segment_dists[perp] = np.abs(
-            segment_coords[perp, 0] * np.cos(segment_angle)
-            + segment_coords[perp, 1] * np.sin(segment_angle)
-            - segment_dist
+            (segment_coords[perp, 0] * dc - segment_coords[perp, 1] * dr \
```

Comment:
I think @stefanv meant that it's likely that the code will raise an exception (division by zero). I guess it's something we can check with a polygon for which a point appears twice and consecutively. That would be a test to add in the test suite anyway.